src/librustdoc/html/render/print_item.rs

```diff
@@ -1663,7 +1663,7 @@ fn document_type_layout(w: &mut Buffer, cx: &Context<'_>, ty_def_id: DefId) {
             writeln!(
                 w,
                 "<div class=\"warning\"><p><strong>Note:</strong> Most layout information is \
-                 completely unstable and may be different between compiler versions and platforms. \
+                 completely unstable and may even differ between compilations. \
```

Comment:
How can this be true? The compiler has to be able to depend on the ABI to create dynamic libraries.

Reply:
From [the Reference](https://doc.rust-lang.org/nightly/reference/type-layout.html#type-layout) (emphasis mine):

> Type layout can be changed with *each compilation*. Instead of trying to document exactly what is done, we only document what is guaranteed today.

I assume that means that the compiler is free to change layout if your crate is a static library or a binary. Also, there are (future) flags like `-Z randomize-layout` which would change layout between compilations.